engine/pkg/externalresource/manager/service.go

```diff
@@ -43,46 +42,65 @@ type Service struct {
 	wg       sync.WaitGroup
 	cancelCh chan struct{}
 
-	offlinedExecutors chan resModel.ExecutorID
-	preRPCHook        rpcutil.PreRPCHook
+	preRPCHook rpcutil.PreRPCHook
 }
 
-const (
-	offlineExecutorQueueSize = 1024
-)
-
 // NewService creates a new externalresource manage service
 func NewService(
 	metaclient pkgOrm.Client,
 	executorInfoProvider ExecutorInfoProvider,
 	preRPCHook rpcutil.PreRPCHook,
 ) *Service {
 	return &Service{
-		metaclient:        metaclient,
-		executors:         executorInfoProvider,
-		offlinedExecutors: make(chan resModel.ExecutorID, offlineExecutorQueueSize),
-		preRPCHook:        preRPCHook,
+		metaclient: metaclient,
+		executors:  executorInfoProvider,
+		preRPCHook: preRPCHook,
 	}
 }
 
 // QueryResource implements ResourceManagerClient.QueryResource
-func (s *Service) QueryResource(ctx context.Context, request *pb.QueryResourceRequest) (*pb.QueryResourceResponse, error) {
+func (s *Service) QueryResource(
+	ctx context.Context,
+	request *pb.QueryResourceRequest,
+) (
+	resp *pb.QueryResourceResponse,
+	retErr error,
+) {
 	var resp2 *pb.QueryResourceResponse
 	shouldRet, err := s.preRPCHook.PreRPC(ctx, request, &resp2)
 	if shouldRet {
 		return resp2, err
 	}
 
-	record, err := s.metaclient.GetResourceByID(ctx,
-		pkgOrm.ResourceKey{JobID: request.GetResourceKey().GetJobId(), ID: request.GetResourceKey().GetResourceId()})
+	defer func() {
+		if retErr != nil {
+			retErr = rpcerror.ToGRPCError(retErr)
```

Comment:
https://github.com/pingcap/tiflow/pull/6669 will add a server interceptor to convert error. So we don't need to convert the error manually.